crypto/x509/x509_lu.c

```diff
@@ -613,13 +615,21 @@ int X509_STORE_CTX_get1_issuer(X509 **issuer, X509_STORE_CTX *ctx, X509 *x) {
       }
       if (ctx->check_issued(ctx, x, pobj->data.x509)) {
         *issuer = pobj->data.x509;
-        X509_OBJECT_up_ref_count(pobj);
         ret = 1;
-        break;
+        // Break the loop with a match if the time check is valid, otherwise
+        // we continue searching. We leave the last tested issuer certificate in
+        // |issuer| on purpose. This returns the closest match if none of the
+        // candidate issuer certificates' timestamps were valid.
+        if (x509_check_cert_time(ctx, *issuer, /*suppress_error*/1)) {
+          break;
```

Comment:
is the comment on L579 still true?

```
// However it will fill the cache with all matching certificates, so we can
// examine the cache for all matches.
```

Reply:
I took a closer look and I think it still holds. The "cache" being mentioned here is `ctx->ctx->objs`, which is being looped over and holds all matching certs.